src/compressed_chunk.c

```diff
@@ -188,13 +188,14 @@ u_int64_t getIterIdx(ChunkIter_t *iter) {
 }
 // LCOV_EXCL_STOP
 
-ChunkIter_t *Compressed_NewChunkIterator(Chunk_t *chunk, bool rev) {
+ChunkIter_t *Compressed_NewChunkIterator(Chunk_t *chunk, int options) {
     CompressedChunk *compressedChunk = chunk;
 
     // for reverse iterator of compressed chunks
-    if (rev == true) {
+    if (options & CHUNK_ITER_OP_REVERSE) {
+        int uncompressed_options = CHUNK_ITER_OP_REVERSE | CHUNK_ITER_OP_FREE_CHUNK;
```

Comment:
@danni-m why do you need to define `uncompressed_options` with `CHUNK_ITER_OP_FREE_CHUNK`? I don't see that this flag is tested in `Uncompressed_NewChunkIterator`.

Why not just?
```
         return Uncompressed_NewChunkIterator(uncompressedChunk, CHUNK_ITER_OP_REVERSE); 
```

Reply:
@gkorland 
B/c some in regular range we don't free as we don't own the chunks
In revenge, we decompressed and therefore we need to free

Reply:
To expand what ariel wrote, an iterator "owns" a chunk (and not the other way around) only when a compressed chunk is iterated in reverse. So logically we should only support that specific scenario as the compressed chunk iterator should never own the actual chunk.